### PR TITLE
Pivotal ID # 176522202: Ext Submission serialization

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/176522202

### CODE REVIEW

File: commons/commons-model-extended-serialization/src/main/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtSubmissionSerializer.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        gen.writeObjectField(TAGS, submission.tags)
+        gen.writeObjectField(PROJECTS, submission.projects)
+        gen.writeObjectField(STATS, submission.stats)
+        gen.writeObjectField(ACCESS_TAGS, getAccessTag(submission))
+        gen.writeEndObject()
+    }
+
+    private fun getAccessTag(submission: ExtSubmission): List<ExtAccessTag> {
+        val projects = submission.projects.map { ExtAccessTag(it.accNo) }.plus(ExtAccessTag(submission.owner))
+        return if (submission.released) projects.plus(ExtAccessTag("Public")) else projects
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended-serialization/src/main/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtSubmissionSerializer.kt:58]: I think we have a constant for this "Public" access tag don't we?

File: commons/commons-model-extended-serialization/src/test/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtSubmissionSerializerTest.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+
+@ExtendWith(TemporaryFolderExtension::class)
+class ExtSubmissionSerializerTest {
+    private val testInstance = ObjectMapper().registerModule(createModule())
+
+    @Test
+    fun `serialize basic section when not released`() {
+        val extendedSubmission = testSubmission(false)
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended-serialization/src/test/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtSubmissionSerializerTest.kt:35]: I think adding the name parameter would make the test clearer, i.e. `testSubmission(released = false)`

File: commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/model/ExtendedModel.kt [hunk 3]
```diff
@@ -93,11 +90,7 @@ data class ExtSubmission(
     val tags: List<ExtTag> = listOf(),
     val projects: List<Project> = listOf(),
     val stats: List<ExtStat> = listOf()
-) {
-    // TODO: add custom serializer/deserializer to avoid json annotation in model
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/model/ExtendedModel.kt:97]: 🤘

**Review by jhoanmanuelms — APPROVED** [review]
LGTM. Please consider my comments.

File: commons/commons-model-extended-serialization/src/main/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtSubmissionSerializer.kt (hunk 1 above)
Comment (Juan-EBI) [reply commons/commons-model-extended-serialization/src/main/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtSubmissionSerializer.kt:58]: yes but it lives in the module commons-bio which contain page tab representation and we do not want to make this module depends on that one.

**Pull request merged**